### PR TITLE
isPassport and isPassportFormat

### PR DESCRIPTION
**Pull request by alepaz**
isPassport(passport, DUI)
In case the passport has a valid format and the DUI is valid

isPassportFormat(passport)
In case the passport has a valid format starts with: [A|B|D] and continues with 8 digit

### CODE REVIEW

File: src/lib/documents.ts [hunk 1]
```diff
@@ -72,3 +73,26 @@ export function isNIT(str: string): boolean {
...
+}
+
+/**
+ * Verifies that given Passport is valid
+ *
+ * @param  {string} str       A string representing the passport
+ * @param  {string} dui       A string representing the dui
+ * @returns {boolean}         Validity of the given passport
+ */
+export function isPassport(str: string, dui: string): boolean {
```
Comment (jonathanpalma) [inline src/lib/documents.ts:171]: Please remove the DUI verification as a requirement to validate passport considering that there are many residents that use their international passports. So I don't think it would be a good idea to create a constraint, based on this parameter.

File: src/lib/documents.ts [hunk 2]
```diff
@@ -3,25 +3,60 @@ import { isDate } from './utils';
...
+}
+
+/**
+ * Verifies that given DUI in numeric format is valid
+ *
+ * @param  {string} str       A string representing DUI digits (without hyphen)
+ * @returns {boolean}         Validity of the given DUI
+ */
+function isDUIWithNumbers(str: string): boolean {
+  if (!(str.length === 9)) return false;
```
Comment (jonathanpalma) [inline src/lib/documents.ts:43]: Can you please:

1. Submit this feature in a separate PR
2. Evaluate the string using RegExp, instead of just validating the length of the string. In this way, we are going to be able to verify that the string only contains digits

File: src/lib/documents.ts [hunk 3]
```diff
@@ -3,25 +3,60 @@ import { isDate } from './utils';
...
+}
+
+/**
+ * Verifies that given DUI format is valid
+ *
+ * @param  {string} str       A string representing DUI digits (with hyphen)
+ * @returns {boolean}         Validity of the given DUI
+ */
+export function isDUI(str: string): boolean {
+  if (Number(str)) return isDUIWithNumbers(str);
```
Comment (jonathanpalma) [inline src/lib/documents.ts:56]: As I mentioned before this feature should be submitted as a separate PR but I would recommend using RegExp instead of relying on `falsy` values (`0`, `NaN`, etc)

File: src/lib/documents.ts [hunk 4]
```diff
@@ -3,25 +3,60 @@ import { isDate } from './utils';
...
+  return isValidDUI(Number(digits), Number(verifier));
+}
+
+/**
+ * Verifies that given DUI in numeric format is valid
+ *
+ * @param  {string} str       A string representing DUI digits (without hyphen)
+ * @returns {boolean}         Validity of the given DUI
+ */
+function isDUIWithNumbers(str: string): boolean {
```
Comment (jonathanpalma) [inline src/lib/documents.ts:42]: Please rename to `isDUIWithoutHyphen`

File: src/lib/documents.ts [hunk 5]
```diff
@@ -72,3 +112,64 @@ export function isNIT(str: string): boolean {
...
+  return isValidNIT(municipality, birthdate, correlative, verifier);
+}
+
+/**
+ * Verifies that given NIT in numeric format is valid
+ *
+ * @param  {string} str       A string representing NIT digits (without hyphens)
+ * @returns {boolean}         Validity of the given NIT
+ */
+function isNITWithNumbers(str: string): boolean {
```
Comment (jonathanpalma) [inline src/lib/documents.ts:134]: Rename to `isNITWithoutHyphen`

File: src/lib/documents.ts [hunk 6]
```diff
@@ -72,3 +112,64 @@ export function isNIT(str: string): boolean {
...
+  if (!(str.length === 14)) return false;
```
Comment (jonathanpalma) [inline src/lib/documents.ts:135]: Can you please:

1. Submit this feature in a separate PR
2. Evaluate the string using RegExp, instead of just validating the length of the string. In this way, we are going to be able to verify that the string only contains digits

File: src/lib/documents.ts [hunk 7]
```diff
@@ -72,3 +112,64 @@ export function isNIT(str: string): boolean {
...
+}
+
+/**
+ * Verifies that given NIT format is valid
+ *
+ * @param  {string} str       A string representing NIT digits (with hyphen)
+ * @returns {boolean}         Validity of the given NIT
+ */
+export function isNIT(str: string): boolean {
+  if (Number(str)) return isNITWithNumbers(str);
```
Comment (jonathanpalma) [inline src/lib/documents.ts:150]: As I mentioned before this feature should be submitted as a separate PR but I would recommend using RegExp instead of relying on `falsy` values (`0`, `NaN`, etc)

File: src/lib/documents.ts (hunk 1 above)
Comment (alepaz) [reply src/lib/documents.ts:171]: Done in a new PR

**Comment by jonathanpalma** [comment]
Closing in favor of #14
